pkg/manager/internal.go

```diff
@@ -260,6 +263,21 @@ func (cm *controllerManager) SetFields(i interface{}) error {
 	return nil
 }
 
+// AddMetricsExtraHandler adds extra handler served on path to the http server that serves metrics.
+func (cm *controllerManager) AddMetricsExtraHandler(path string, handler http.Handler) error {
+	cm.mu.Lock()
+	defer cm.mu.Unlock()
+
+	_, found := cm.metricsExtraHandlers[path]
```

Comment:
This allows to overwrite the builtin metrics endpoint as its not in the `metricsExtraHandlers` map or did I miss something?

Reply:
@alvaroaleman  Yeah, I missed this point, thanks. If we try to override `/metrics` endpoint, `serveMetrics` will actually panic.

What could you suggest here? Options that come to my mind:
- When adding extra handler, if we see `/metrics`, we log this attempt now, ignoring this overriden endpoint
- When trying to register `/metrics` override in `AddMetricsExtraHandler`, we could return an error saying this is not allowed.

Also we need to document this nuance on `AddMetricsExtraHandler` method.

Reply:
Its a bit unfortunate that this can only be discovered at runtime. IMHO we should return an error, otherwise we probably hide bugs and keep ppl wondering why their metrics do not appear.

Reply:
Added the check to `AddMetricsExtraHandler`, PTAL

Reply:
thanks